apps/android/firefox.py

```diff
@@ -74,28 +83,95 @@ def start(self, url="about:blank"):
             }
         )
 
-        device.push(profile.profile, device_profile)
-        device.chmod(device_storage, recursive=True)
+        self.device.push(self.profile.profile, device_profile)
+        self.device.chmod(device_storage, recursive=True)
 
-        app_args = [
-            "-profile",
-            device_profile,
-            "--marionette",
-            "--es",
-            "env0",
-            "LOG_VERBOSE=1",
-            "--es",
-            "env1",
-            "R_LOG_LEVEL=6",
-        ]
+    def run_android_app(self, url):
+        raise NotImplementedError
 
+    def start(self, url="about:blank"):
+        # create profile
+        self.set_profile()
+        # create certificate database
+        self.create_certificate()
+        # setup device
+        self.setup_device()
         # start app
-        device.stop_application(self.APP_NAME)
-        device.launch_activity(
+        self.run_android_app(url)
+
+
+class GeckoViewExample(AbstractAndroidFirefox):
+    APP_NAME = "org.mozilla.geckoview_example"
+    ACTIVITY_NAME = "GeckoViewActivity"
+
+    def run_android_app(self, url):
+        self.device.stop_application(self.APP_NAME)
+        self.device.launch_activity(
             self.APP_NAME,
             self.ACTIVITY_NAME,
-            extra_args=app_args,
+            extra_args=self.app_args,
             url=url,
             e10s=True,
-            fail_if_running=False,
+            fail_if_running=False
+        )
+
+
+class Fenix(AbstractAndroidFirefox):
+    APP_NAME = "org.mozilla.fenix.raptor"
+    ACTIVITY_NAME = "org.mozilla.fenix.browser.BrowserPerformanceTestActivity"
+    INTENT = "android.intent.action.VIEW"
+
+    def run_android_app(self, url):
+        extras = {
+            "args": " ".join(self.app_args),
+            "use_multiprocess": True
+        }
+
+        # start app
+        self.device.stop_application(self.APP_NAME)
+        self.device.launch_application(
+            self.APP_NAME,
+            self.ACTIVITY_NAME,
+            self.INTENT,
+            extras=extras,
+            url=url,
+            fail_if_running=False
+        )
+
+
+class Fennec(AbstractAndroidFirefox):
+    APP_NAME = "org.mozilla.fennec_aurora"
+    ACTIVITY_NAME = "org.mozilla.gecko.BrowserApp"
+    INTENT = "android.intent.action.VIEW"
+
+    def run_android_app(self, url):
+        self.device.stop_application(self.APP_NAME)
+        self.device.launch_fennec(
+            self.APP_NAME,
+            extra_args=self.app_args,
+            url=url,
+            fail_if_running=False
+        )
+
+
+class RefBrow(AbstractAndroidFirefox):
+    APP_NAME = "org.mozilla.reference.browser"
+    ACTIVITY_NAME = "org.mozilla.reference.browser.BrowserTestActivity"
+    INTENT = "android.intent.action.MAIN"
+
+    def run_android_app(self, url):
+        extras = {
```

Comment:
move this as a class property

Reply:
`GeckoViewExample` doesn't have it. Do you want to make it property for each subclass?